tutorials-and-examples/llamaindex/rag/README.md

```diff
@@ -0,0 +1,481 @@
+  * `embed_model`: an embedding model that generates vector representations for the processed text.  
+* `docstore`: defines where the processed documents are stored for later retrieval.  
+* `vector_store`: a place where the resulting nodes are placed.  
+* `cache`: a place where to store a cache for a faster document search.  
+* `docstore_strategy`: defines how the document store handles incoming data. In our case, it updates existing documents or inserts new ones if they don’t already exist.
+</details>
+
+File `app/rag_demo/__init__.py`:  
+
+<details><summary>Expand to see key parts</summary>
+
+
+In this demo we use [Redis-stack](https://redis.io/about/about-stack/) as our vector storage and we also have to define a data schema for it.
+
+```
+custom_schema = IndexSchema.from_dict(
+    {
+        "index": {"name": "bucket", "prefix": "doc"},
+        # customize fields that are indexed
+        "fields": [
+            # required fields for llamaindex
+            {"type": "tag", "name": "id"},
+            {"type": "tag", "name": "doc_id"},
+            {"type": "text", "name": "text"},
+            # custom vector field for bge-small-en-v1.5 embeddings
+            {
+                "type": "vector",
+                "name": "vector",
+                "attrs": {
+                    "dims": 384,
+                    "algorithm": "hnsw",
+                    "distance_metric": "cosine",
+                },
+            },
+        ],
+    }
+)
+```
+</details>
+
+### RAG server
+
+Simple web application that invokes llamaindex RAG system and returns response. The web application will be a simple FastAPI app, so we can invoke the RAG system via HTTP request. It is defined in the `app/rag_demo/main.py` file as a FastAPI application and a single `/invoke` API and connection to the Redis Vector Store.
+
+<details><summary>Expand to see key parts</summary>
+
+```
+embed_model = HuggingFaceEmbedding(model_name=EMBEDDING_MODEL_NAME)
+
+# Connect to vector store with already ingested data
+vector_store = RedisVectorStore(
+    schema=custom_schema,
+    redis_url=f"redis://{REDIS_HOST}:{REDIS_PORT}",
+)
+
+# Create index from a vector store
+index = VectorStoreIndex.from_vector_store(
+    vector_store, embed_model=embed_model
+)
+
+# Connect to LLM using Ollama
+llm = Ollama(
+    model=MODEL_NAME,
+    base_url=OLLAMA_SERVER_URL,
+)
+
+# Create query engine that is ready to query our RAG
+query_engine = index.as_query_engine(llm=llm)
+```
+
+</details>
+
+### 
+
+# Before you begin
+
+Ensure you have a GCP project with a billing account.
+
+Ensure you have the following tools installed on your workstation:
+
+* [gcloud CLI](https://cloud.google.com/sdk/docs/install)  
+* [gcloud kubectl](https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-access-for-kubectl#install_kubectl)  
```

Comment:
minor nit: Can we change the title `gcloud kubectl` to `Install kubectl` for example ? kubectl isn't really part of gcloud CLI

Reply:
What about just `kubectl`? Just to keep it consistent with other items from the list which are all nouns.

Reply:
Sounds good!